include/mlir/SPIRV/SPIRVTypes.h

```diff
@@ -52,9 +53,28 @@ enum Kind {
 };
 }
 
+// Provides a common base class between SPIR-V ArrayType, SPIR-V StructType.
+class CompositeType : public Type {
+public:
+  using ImplType = detail::CompositeTypeStorage;
+  using Type::Type;
+
+  static bool kindof(unsigned kind) {
+    return (kind == TypeKind::Array || kind == TypeKind::Struct);
```

Comment:
Oh, I see, to be able to use kinda "cast" thing we have to provide "kindof" function. I was stuck for I a while trying to understand how to use Vector from Standard Types as kinda CompositeType, in case we can't directly modify it. Thanks a lot, now I'm understanding MLIR's type system more better. Sorry, for this kind of realization.